sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/test/TestTableProvider.java

```diff
@@ -150,12 +153,30 @@ public BeamTableStatistics getTableStatistics(PipelineOptions options) {
       return begin.apply(Create.of(tableWithRows.rows).withCoder(rowCoder()));
     }
 
+    @Override
+    public PCollection<Row> buildIOReader(
+        PBegin begin, BeamSqlTableFilter filters, List<String> fieldNames) {
+      Preconditions.checkNotNull(fieldNames);
```

Comment:
In `BaseBeamTable` it is acceptable for fieldNames to be null. You should be consistent in what you accept. Should that interface also be checking that these are not null?

Reply:
Decided not to support `null` values as arguments.